### PR TITLE
issue #7 idea comments

### PR DESCRIPTION
**Pull request by pmeads**
Please review this code for the idea comments.  used awesome_nested_set gem to add nested comments for ideas. created a comments table and used an ajax request to create the reply comment form. the jquery code is in blocks/componets/comments.js . I wasn't sure if we were using coffeescript or not (i didn't).

at some point, performance should be tested for a big comments table as i see one of the gem's update statements requires a full table scan.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      raise Application::Error.new "You must be logged in to comment",
+                                     redirect_to: [
+                                         auth_oauth2_launch_url(:shibboleth),
+                                         flash: { return_to: path }
+                                     ]
+    end
+
+    comment = Comment.new params[:comment].permit(:parent_id, :idea_id, :body, :commit)  
+    comment.user_id = context.user.id
+    comment.save
```
Comment (chabgood) [inline app/controllers/comments_controller.rb]: You can do context.user.comments.create(params[:comment].permit(:parent_id, :idea_id, :body, :commit))

Comment (pmeads) [inline app/controllers/comments_controller.rb]: Thanks for the tip! That also helped me discover that the User model was missing the relation to Comment. Just committed the new code.

**Comment by pmeads** [comment]
I just did the following from my workstation:
git fetch upstream
git merge upstream/master
git checkout idea_nested_comments_7
git merge master
vi db/schema.rb
vi Gemfile
git add Gemfile
git add db/schema.rb 
git commit -m "Merged master and fixed conflicts"
git push origin idea_nested_comments_7

I think that should have merged the master branch with the branch idea_nested_comments_7. I did have to resolve two conflicts to complete the merge. I then pushed that to github. Is there anything else I need to do?

**Comment by chabgood** [comment]
You might do a rebase when merging master into your branch.  But nothing
else from what I can see.

On Thu, Jun 4, 2015 at 12:03 PM, Philip Meads <notifications@github.com>
wrote:

> I just did the following from my workstation:
> git fetch upstream
> git merge upstream/master
> git checkout idea_nested_comments_7
> git merge master
> vi db/schema.rb
> vi Gemfile
> git add Gemfile
> git add db/schema.rb
> git commit -m "Merged master and fixed conflicts"
> git push origin idea_nested_comments_7
>
> I think that should have merged the master branch with the branch
> idea_nested_comments_7. I did have to resolve two conflicts to complete the
> merge. I then pushed that to github. Is there anything else I need to do?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/universityofcalifornia/NeXt/pull/71#issuecomment-109011387>
> .
>



-- 


*"In matters of style, swim with the current; in matters of principle,
stand like a rock."  Thomas Jefferson*

**Comment by pmeads** [comment]
The Travis CI build failed its tests b/c I defined "comments.user_id" as a not null column with the assumption that only people who logged in can comment.

**Comment by chabgood** [comment]
You can do "bundle exec rake test" from your cli which is what travis runs.


On Thu, Jun 4, 2015 at 12:08 PM, Philip Meads <notifications@github.com>
wrote:

> The Travis CI build failed its tests b/c I defined "comments.user_id" as a
> not null column with the assumption that only people who logged in can
> comment.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/universityofcalifornia/NeXt/pull/71#issuecomment-109013919>
> .
>



-- 


*"In matters of style, swim with the current; in matters of principle,
stand like a rock."  Thomas Jefferson*

**Comment by pmeads** [comment]
Cool, thx. I'll run the tests locally. Off the top of my head, I'm not sure how to get the tests to log in to as a user and start a session. It looks like there is a users fixture i can use but not sure how to do it yet.

**Comment by chabgood** [comment]
Try to use factories and not fixtures.  I implemented rspec which has
factory girl in use.  I have yet to be able to get capybara to log in yet.

On Thu, Jun 4, 2015 at 1:07 PM, Philip Meads <notifications@github.com>
wrote:

> Cool, thx. I'll run the tests locally. Off the top of my head, I'm not
> sure how to get the tests to log in to as a user and start a session. It
> looks like there is a users fixture i can use but not sure how to do it
> yet.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/universityofcalifornia/NeXt/pull/71#issuecomment-109033678>
> .
>



-- 


*"In matters of style, swim with the current; in matters of principle,
stand like a rock."  Thomas Jefferson*

**Comment by pmeads** [comment]
ok, i'll look at what you did as an example. thx!

**Comment by chabgood** [comment]
Sorry I did not see this earlier,  the comments table has more than 1 responsiblity?

**Comment by pmeads** [comment]
i thought maybe we should leave it open in case we wanted projects to have
comments as well

On Thu, Jun 11, 2015 at 3:34 PM, Chris Habgood <notifications@github.com>
wrote:

> Sorry I did not see this earlier, the comments table has more than 1
> responsiblity?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/universityofcalifornia/NeXt/pull/71#issuecomment-111294849>
> .
>



-- 
Philip Meads
Database Administrator
UCSC
831 459 1705
philme@ucsc.edu

**Comment by rrocchio** [comment]
@pmeads  - yes - lets add project comments!  It was suggested on UC Next!

**Pull request merged**